analysis/src/python/data_analysis/template_errors/search.py

```diff
@@ -0,0 +1,113 @@
+import argparse
+import json
+import sys
+from typing import Callable, List, Tuple
+
+import pandas as pd
+
+from analysis.src.python.data_analysis.model.column_name import SubmissionColumns
+from analysis.src.python.data_analysis.utils.df_utils import read_df
+from analysis.src.python.evaluation.issues_statistics.common.raw_issue_encoder_decoder import RawIssueDecoder
+
+
+def submission_to_issue_code_pairs(submission: pd.DataFrame) -> List[Tuple[str, str]]:
+    """
+    Represent issue as a pair <origin_class, line>.
+    """
+    issues = submission[SubmissionColumns.RAW_ISSUES.value]
+    return [(issue.origin_class, submission[SubmissionColumns.CODE.value][issue.line_no - 1]) for issue in issues]
+
+
+equal_lines: Callable[[str, str], bool] = lambda x, y: x == y
+
+
+def equal_issues(issue: Tuple[str, str], other_issue: Tuple[str, str]) -> bool:
+    return issue[0] == other_issue[0] and equal_lines(issue[1], other_issue[1])
+
+
+def get_repetitive_issues(issues_by_attempt: List[List[Tuple[str, str]]]) -> List[Tuple[str, str]]:
+    """
+    Find issues that occur in every attempt of the series issues_by_attempt.
+    """
+
+    def is_in_previous_issues(issue_line: Tuple[str, str],
+                              previous_issues_lines: List[Tuple[str, str]]) -> bool:
+        for pr_issue_line in previous_issues_lines:
+            if equal_issues(issue_line, pr_issue_line):
+                return True
+        return False
```

Comment:
can we replace it with [any](https://docs.python.org/3/library/functions.html#any) function?